support/util/imagefetch.go

```diff
@@ -0,0 +1,63 @@
+package util
+
+import (
+	"context"
+	"fmt"
+	"github.com/blang/semver/v4"
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/support/releaseinfo"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+// GetHypershiftComponentImage resolves the appropriate hypershift
+// image based on the following order of precedence (from most to least
+// preferred):
+//
+// 1. The image specified by the ControlPlaneOperatorImageAnnotation on the
+//    HostedControlPlane resource itself
+// 2. The hypershift image specified in the release payload indicated by the
+//    HostedControlPlane's release field
+// 3. The hypershift-operator's own image for release versions 4.9 and 4.10
+//
+// If no image can be found according to these rules, an error is returned.
+func GetHypershiftComponentImage(ctx context.Context, objectAnnotations map[string]string, releaseImage string, releaseProvider releaseinfo.Provider, hypershiftOperatorImage string, pullSecret []byte) (string, error) {
+	if val, ok := objectAnnotations[hyperv1.ControlPlaneOperatorImageAnnotation]; ok {
+		return val, nil
+	}
+	releaseInfo, err := releaseProvider.Lookup(ctx, releaseImage, pullSecret)
+	if err != nil {
+		return "", err
+	}
+	version, err := semver.Parse(releaseInfo.Version())
+	if err != nil {
+		return "", err
+	}
+
+	if hypershiftImage, exists := releaseInfo.ComponentImages()["hypershift"]; exists {
+		return hypershiftImage, nil
+	}
+
+	versionMajMin := fmt.Sprintf("%d.%d", version.Major, version.Minor)
+	switch versionMajMin {
+	case "4.9", "4.10":
+		return hypershiftOperatorImage, nil
+	default:
+		return "", fmt.Errorf("unsupported release image with version %s", versionMajMin)
+	}
+}
+
+// LookupActiveContainerImage returns the image for a specified container in a specified pod
+func LookupActiveContainerImage(ctx context.Context, pods v1.PodInterface, podName string, containerName string) (string, error) {
```

Comment:
Please use the controller runtime client like we do everywhere else in order to be consistent and avoid an additional ~20k LOC of deps. It will by default create a watch for all pods which we don't want, but we can set up a filter so that it will only watch its own pod, which is actually more efficient than making api requests here

Reply:
If https://github.com/openshift/hypershift/pull/963#discussion_r801840526 is possible, doesn't this all go away?

Reply:
I think maybe @alvaroaleman already mentioned this, could this be downward API?

Reply:
Downward API can't expose the container image directly, but it could inject from an annotation or label, which might be worth it considering the alternative (the client stuff here)?

Reply:
No the dependency is added because we need the fake client set for testing of the standard client-go Kube Client. That is already utilized throughout the code and this provides a way to unit test it. If we want it gone: we would need to remove all references of that

Reply:
FYI pod lookup will break the native process workflows (I don't use them anymore but others might)